src/v/cloud_storage/remote_partition.h

```diff
@@ -95,7 +96,17 @@ class remote_partition
     bool bounds_timestamp(model::timestamp) const;
 
     /// Return first uploaded kafka offset
-    kafka::offset first_uploaded_offset();
+    kafka::offset first_uploaded_offset() {
```

Comment:
same question here, in order to move these checks to the header you have to include more stuff in the header which may have impact on the build